protocol/rep/rep.go

```diff
@@ -227,9 +225,8 @@ func (r *rep) RecvHook(m *mangos.Message) bool {
 	}
 	r.sock.SetSendError(nil)
 	r.backtraceL.Lock()
-	r.backtrace = append(r.backtracebuf[0:0], m.Header...)
+	r.backtrace = append(r.backtrace[0:0], m.Header...)
 	r.backtraceL.Unlock()
-	m.Header = nil
```

Comment:
Fully agree - the SP 31-bit request IDs are more likely to have a collision than the [128 bit variants](https://tools.ietf.org/html/rfc4122). 

Having to re-implement that on top of SP would duplicate effort, likely there would be another protocol header needed to wrap in the UUID; adding another layer on top of SP.

What is the best way forward with this; should I be writing an issue or similar?